## ExoQuantum AI

Este notebook tem como objetivo validar o dataset escolhido para o projeto **ExoQuantum AI**, que propõe a classificação e priorização de candidatos a exoplanetas utilizando técnicas de Machine Learning, Deep Learning e, posteriormente, Quantum Machine Learning.

Nesta etapa, o foco não é treinar modelos, mas verificar se os dados disponíveis são adequados para o problema proposto.

Serão analisados:

- origem do dataset;
- formato de acesso;
- quantidade de registros;
- colunas disponíveis;
- variável-alvo;
- distribuição das classes;
- valores ausentes;
- variáveis numéricas relevantes;
- viabilidade inicial para modelagem.

## 1. Contexto do dataset

O dataset utilizado vem do **NASA Exoplanet Archive**, uma base mantida pela NASA/IPAC que reúne dados relacionados a exoplanetas, candidatos planetários e observações astronômicas.

Neste projeto, será utilizada a tabela **Kepler Objects of Interest (KOI)**. Um KOI é um objeto observado pela missão Kepler que apresentou sinais compatíveis com a possível presença de um exoplaneta, geralmente por meio do método de trânsito planetário.

A tabela contém objetos classificados em categorias como:

- `CONFIRMED`: planeta confirmado;
- `CANDIDATE`: candidato a exoplaneta;
- `FALSE POSITIVE`: sinal que parecia indicar um planeta, mas foi classificado como falso positivo.

Essa estrutura é adequada para um problema de classificação supervisionada, pois permite treinar modelos para distinguir candidatos promissores de falsos positivos com base em características observacionais.

In [6]:
%pip install pandas numpy

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 9.9/9.9 MB 47.5 MB/s eta 0:00:0000:01
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 5.2/5.2 MB 48.4 MB/s eta 0:00:00

[notice] A new release of pip is available: 24.3.1 -> 26.1.2
[notice] To update, run: pip3 install --upgrade pip
Note: you may need to restart the kernel to use updated packages.


In [7]:
import pandas as pd
import numpy as np

pd.set_option("display.max_columns", None)
pd.set_option("display.max_rows", 100)

In [9]:
%pip install certifi

  Using cached certifi-2026.5.20-py3-none-any.whl.metadata (2.5 kB)
Using cached certifi-2026.5.20-py3-none-any.whl (134 kB)

[notice] A new release of pip is available: 24.3.1 -> 26.1.2
[notice] To update, run: pip3 install --upgrade pip
Note: you may need to restart the kernel to use updated packages.


In [12]:
from urllib.parse import quote

base_url = "https://exoplanetarchive.ipac.caltech.edu/TAP/sync"

query = "select * from cumulative"

url = f"{base_url}?query={quote(query)}&format=csv"

In [13]:
import ssl
import certifi
import urllib.request

ssl_context = ssl.create_default_context(cafile=certifi.where())

response = urllib.request.urlopen(url, context=ssl_context)

df = pd.read_csv(response)

df.head()

,kepid,kepoi_name,kepler_name,ra,ra_err,ra_str,dec,dec_err,dec_str,koi_gmag,koi_gmag_err,koi_rmag,koi_rmag_err,koi_imag,koi_imag_err,koi_zmag,koi_zmag_err,koi_jmag,koi_jmag_err,koi_hmag,koi_hmag_err,koi_kmag,koi_kmag_err,koi_kepmag,koi_kepmag_err,koi_delivname,koi_vet_stat,koi_quarters,koi_disposition,koi_pdisposition,koi_count,koi_num_transits,koi_max_sngle_ev,koi_max_mult_ev,koi_bin_oedp_sig,koi_limbdark_mod,koi_ldm_coeff4,koi_ldm_coeff3,koi_ldm_coeff2,koi_ldm_coeff1,koi_trans_mod,koi_model_snr,koi_model_dof,koi_model_chisq,koi_time0bk,koi_time0bk_err1,koi_time0bk_err2,koi_eccen,koi_eccen_err1,koi_eccen_err2,koi_longp,koi_longp_err1,koi_longp_err2,koi_prad,koi_prad_err1,koi_prad_err2,koi_sma,koi_sma_err1,koi_sma_err2,koi_impact,koi_impact_err1,koi_impact_err2,koi_duration,koi_duration_err1,koi_duration_err2,koi_ingress,koi_ingress_err1,koi_ingress_err2,koi_depth,koi_depth_err1,koi_depth_err2,koi_period,koi_period_err1,koi_period_err2,koi_ror,koi_ror_err1,koi_ror_err2,koi_dor,koi_dor_err1,koi_dor_err2,koi_incl,koi_incl_err1,koi_incl_err2,koi_teq,koi_teq_err1,koi_teq_err2,koi_steff,koi_steff_err1,koi_steff_err2,koi_slogg,koi_slogg_err1,koi_slogg_err2,koi_smet,koi_smet_err1,koi_smet_err2,koi_srad,koi_srad_err1,koi_srad_err2,koi_smass,koi_smass_err1,koi_smass_err2,koi_sage,koi_sage_err1,koi_sage_err2,koi_sparprov,koi_fwm_stat_sig,koi_fwm_sra,koi_fwm_sra_err,koi_fwm_sdec,koi_fwm_sdec_err,koi_fwm_srao,koi_fwm_srao_err,koi_fwm_sdeco,koi_fwm_sdeco_err,koi_fwm_prao,koi_fwm_prao_err,koi_fwm_pdeco,koi_fwm_pdeco_err,koi_dicco_mra,koi_dicco_mra_err,koi_dicco_mdec,koi_dicco_mdec_err,koi_dicco_msky,koi_dicco_msky_err,koi_dikco_mra,koi_dikco_mra_err,koi_dikco_mdec,koi_dikco_mdec_err,koi_dikco_msky,koi_dikco_msky_err,koi_comment,koi_vet_date,koi_tce_plnt_num,koi_tce_delivname,koi_datalink_dvs,koi_disp_prov,koi_parm_prov,koi_time0,koi_time0_err1,koi_time0_err2,koi_datalink_dvr,koi_fpflag_nt,koi_fpflag_ss,koi_fpflag_co,koi_fpflag_ec,koi_insol,koi_insol_err1,koi_insol_err2,koi_srho,koi_srho_err1,koi_srho_err2,koi_fittype,koi_score
0,10797460,K00752.01,Kepler-227 b,291.93423,0.0,19h27m44.22s,48.141651,0.0,+48d08m29.9s,15.890,NaN,15.270,NaN,15.114,NaN,15.006,NaN,14.082,0.025,13.751,0.030,13.648,0.054,15.347,NaN,cumulative,Done,11111111111111111000000000000000,CONFIRMED,CANDIDATE,2,142.0,5.135849,28.470820,0.6864,Claret (2011 A&A 529 75) ATLAS LS,0.0,0.0,0.2291,0.4603,Mandel and Agol (2002 ApJ 580 171),35.8,NaN,NaN,170.538750,0.002160,-0.002160,0.0,NaN,NaN,NaN,NaN,NaN,2.26,0.26,-0.15,0.0853,NaN,NaN,0.146,0.318,-0.146,2.95750,0.08190,-0.08190,NaN,NaN,NaN,615.8,19.5,-19.5,9.488036,2.775000e-05,-2.775000e-05,0.022344,0.000832,-0.000528,24.810,2.600,-2.600,89.66,NaN,NaN,793.0,NaN,NaN,5455.0,81.0,-81.0,4.467,0.064,-0.096,0.14,0.15,-0.15,0.927,0.105,-0.061,0.919,0.052,-0.046,NaN,NaN,NaN,q1_q17_dr25_stellar,0.002,19.462294,1.400000e-05,48.14191,0.000130,0.430,0.510,0.940,0.480,-0.00020,0.00032,-0.00055,0.00031,-0.010,0.130,0.200,0.160,0.200,0.170,0.080,0.130,0.310,0.170,0.320,0.160,NO_COMMENT,2018-08-16 00:00:00,1.0,q1_q17_dr25_tce,010/010797/010797460/dv/kplr010797460-001-2016...,q1_q17_dr25_sup_koi,q1_q17_dr25_koi,2455003.539,0.002160,-0.002160,010/010797/010797460/dv/kplr010797460-20160209...,0,0,0,0,93.59,29.45,-16.65,3.20796,0.33173,-1.09986,LS+MCMC,1.000
1,10797460,K00752.02,Kepler-227 c,291.93423,0.0,19h27m44.22s,48.141651,0.0,+48d08m29.9s,15.890,NaN,15.270,NaN,15.114,NaN,15.006,NaN,14.082,0.025,13.751,0.030,13.648,0.054,15.347,NaN,cumulative,Done,11111111111111111000000000000000,CONFIRMED,CANDIDATE,2,25.0,7.027669,20.109507,0.0023,Claret (2011 A&A 529 75) ATLAS LS,0.0,0.0,0.2291,0.4603,Mandel and Agol (2002 ApJ 580 171),25.8,NaN,NaN,162.513840,0.003520,-0.003520,0.0,NaN,NaN,NaN,NaN,NaN,2.83,0.32,-0.19,0.2734,NaN,NaN,0.586,0.059,-0.443,4.50700,0.11600,-0.11600,NaN,NaN,NaN,874.8,35.5,-35.5,54.418383,2.479000e-04,-2.479000e-04,0.027954,0.009078,-0.001347,77.900,28.400,-28.400,89.57,NaN,NaN,443.0,NaN,NaN,5455.0,81.0,-81.0,4.467,0.064,-0.096,0.14,0.15

In [14]:
df.shape

(9564, 153)

In [15]:
df["koi_disposition"].value_counts(dropna=False)

koi_disposition
FALSE POSITIVE    4839
CONFIRMED         2747
CANDIDATE         1978
Name: count, dtype: int64

### Interpretação da variável-alvo

A coluna `koi_disposition` mostra a classificação de cada objeto observado pela missão Kepler.

No dataset, temos três tipos principais:

- `FALSE POSITIVE`: sinais que pareciam indicar um planeta, mas depois foram classificados como falsos positivos;
- `CONFIRMED`: objetos já confirmados como exoplanetas;
- `CANDIDATE`: objetos que ainda são candidatos e precisam de mais análise.

A classe com mais registros é `FALSE POSITIVE`, com 4.839 exemplos. Depois temos `CONFIRMED`, com 2.747 registros, e `CANDIDATE`, com 1.978 registros.

Essa distribuição mostra que o dataset é adequado para o nosso projeto, porque ele possui exemplos suficientes das três classes. Também percebemos que existe um certo desbalanceamento entre elas, então, nas próximas etapas, não vamos avaliar os modelos apenas pela acurácia. Também vamos usar métricas como precision, recall e F1-score, que ajudam a entender melhor os acertos e erros de cada classe.

In [17]:
columns_df = pd.DataFrame({
    "column_name": df.columns
})

columns_df.head(50)

,column_name
0,kepid
1,kepoi_name
2,kepler_name
3,ra
4,ra_err
5,ra_str
6,dec
7,dec_err
8,dec_str
9,koi_gmag


### Análise inicial das colunas

A primeira inspeção das colunas mostra que o dataset reúne informações de identificação dos objetos, posição no céu, brilho observado e medidas relacionadas ao sinal detectado pela missão Kepler.

A coluna `koi_disposition` será a principal referência para o problema, pois indica a classificação de cada objeto como `CONFIRMED`, `CANDIDATE` ou `FALSE POSITIVE`.

Também aparecem variáveis que podem ser relevantes para a modelagem, como `koi_num_transits` e `koi_model_snr`. A primeira indica a quantidade de trânsitos observados, enquanto a segunda representa a força do sinal em relação ao ruído. Essas informações são importantes porque ajudam a avaliar se o sinal detectado possui consistência suficiente para ser analisado como possível exoplaneta.

Nesta etapa, a análise ainda é exploratória. A seleção final das variáveis será feita depois de avaliarmos a completude dos dados, o significado das colunas e sua relação com a variável-alvo.

In [18]:
columns_df.iloc[50:100]

,column_name
50,koi_longp
51,koi_longp_err1
52,koi_longp_err2
53,koi_prad
54,koi_prad_err1
55,koi_prad_err2
56,koi_sma
57,koi_sma_err1
58,koi_sma_err2
59,koi_impact


### Variáveis físicas e observacionais encontradas

Nesta parte do dataset aparecem variáveis mais diretamente ligadas ao problema de identificação de exoplanetas.

Entre elas, temos `koi_prad`, que representa o raio estimado do possível planeta, `koi_duration`, que indica a duração do trânsito, `koi_depth`, que representa a profundidade da queda de brilho, e `koi_period`, que indica o período orbital estimado.

Também aparecem variáveis relacionadas à estrela hospedeira, como `koi_steff`, `koi_slogg`, `koi_srad` e `koi_smass`. Essas informações são relevantes porque o sinal observado depende não apenas do possível planeta, mas também das características da estrela ao redor da qual ele orbita.

Essas colunas reforçam que o dataset tem atributos suficientes para construir modelos de classificação e, posteriormente, investigar quais características ajudam a diferenciar candidatos, planetas confirmados e falsos positivos.

In [19]:
missing_values = (
    df.isna()
    .sum()
    .sort_values(ascending=False)
    .reset_index()
)

missing_values.columns = ["column_name", "missing_count"]

missing_values["missing_percentage"] = (
    missing_values["missing_count"] / len(df) * 100
).round(2)

missing_values.head(30)

,column_name,missing_count,missing_percentage
0,koi_sma_err2,9564,100.00
1,koi_incl_err2,9564,100.00
2,koi_kepmag_err,9564,100.00
3,koi_incl_err1,9564,100.00
4,koi_model_dof,9564,100.00
5,koi_model_chisq,9564,100.00
6,koi_ingress_err2,9564,100.00
7,koi_sage_err2,9564,100.00
8,koi_eccen_err1,9564,100.00
9,koi_eccen_err2,9564,100.00


### Análise inicial de valores ausentes

A análise de valores ausentes mostra que algumas colunas possuem 100% dos registros vazios. Essas variáveis não trazem informação útil para o projeto e, por isso, deverão ser removidas na etapa de limpeza dos dados.

Também existem colunas com valores ausentes em menor quantidade, como `koi_num_transits`, que possui cerca de 11,94% de registros vazios. Esse tipo de caso ainda pode ser tratado, dependendo da importância da variável para o problema.

Neste momento, a análise indica que será necessário fazer uma seleção cuidadosa das colunas antes da modelagem, priorizando variáveis com bom preenchimento e significado relevante para a classificação dos objetos.

In [20]:
missing_values.tail(40)

,column_name,missing_count,missing_percentage
113,koi_insol_err1,321,3.36
114,koi_srho_err2,321,3.36
115,koi_srho_err1,321,3.36
116,koi_srho,321,3.36
117,koi_insol_err2,321,3.36
118,koi_kmag_err,193,2.02
119,koi_imag,154,1.61
120,koi_hmag_err,97,1.01
121,koi_jmag_err,72,0.75
122,koi_gmag,41,0.43


### Colunas com bom preenchimento

A análise das colunas com menos valores ausentes mostra que algumas variáveis importantes estão completas ou quase completas no dataset.

Entre elas, estão `koi_disposition`, que será usada como variável-alvo, além de variáveis como `koi_period` e `koi_duration`, que descrevem características do trânsito observado.

Também existem colunas de identificação e posição, como `kepid`, `kepoi_name`, `ra` e `dec`, que ajudam a localizar e reconhecer os objetos, mas não necessariamente serão usadas como variáveis principais para o modelo.

Um ponto de atenção é a presença de colunas como `koi_fpflag_ec`, `koi_fpflag_co`, `koi_fpflag_ss` e `koi_fpflag_nt`. Essas colunas parecem estar diretamente relacionadas à identificação de falsos positivos. Por isso, elas devem ser analisadas com cuidado para evitar **data leakage**, ou seja, vazamento de informação da resposta para as variáveis de entrada. Se isso acontecer, o modelo pode usar atalhos e apresentar uma avaliação artificialmente boa.

Com isso, a próxima etapa será selecionar um primeiro conjunto de variáveis candidatas para análise, priorizando colunas com bom preenchimento, significado relevante para o problema e baixo risco de vazamento de dados.

In [21]:
selected_columns = [
    # Identificação
    "kepid",
    "kepoi_name",

    # Variável-alvo
    "koi_disposition",

    # Características do trânsito
    "koi_period",
    "koi_duration",
    "koi_depth",
    "koi_num_transits",
    "koi_model_snr",

    # Características estimadas do possível planeta
    "koi_prad",
    "koi_teq",
    "koi_insol",

    # Características da estrela hospedeira
    "koi_steff",
    "koi_slogg",
    "koi_srad",
    "koi_smass",

    # Observação/brilho
    "koi_kepmag",

    # Posição no céu
    "ra",
    "dec"
]

df_selected = df[selected_columns].copy()

df_selected.head()

,kepid,kepoi_name,koi_disposition,koi_period,koi_duration,koi_depth,koi_num_transits,koi_model_snr,koi_prad,koi_teq,koi_insol,koi_steff,koi_slogg,koi_srad,koi_smass,koi_kepmag,ra,dec
0,10797460,K00752.01,CONFIRMED,9.488036,2.95750,615.8,142.0,35.8,2.26,793.0,93.59,5455.0,4.467,0.927,0.919,15.347,291.93423,48.141651
1,10797460,K00752.02,CONFIRMED,54.418383,4.50700,874.8,25.0,25.8,2.83,443.0,9.11,5455.0,4.467,0.927,0.919,15.347,291.93423,48.141651
2,10811496,K00753.01,CANDIDATE,19.899140,1.78220,10829.0,56.0,76.3,14.60,638.0,39.30,5853.0,4.544,0.868,0.961,15.436,297.00482,48.134129
3,10848459,K00754.01,FALSE POSITIVE,1.736952,2.40641,8079.2,621.0,505.6,33.46,1395.0,891.96,5805.0,4.564,0.791,0.836,15.597,285.53461,48.285210
4,10854555,K00755.01,CONFIRMED,2.525592,1.65450,603.3,515.0,40.9,2.75,1406.0,926.16,6031.0,4.438,1.046,1.095,15.509,288.75488,48.226200


### Análise da seleção inicial de variáveis

O novo dataframe reúne as colunas mais importantes para a primeira versão da análise. Foram mantidas variáveis de identificação, a variável-alvo `koi_disposition` e características físicas e observacionais relacionadas ao trânsito, ao possível planeta e à estrela hospedeira.

Essa seleção ajuda a reduzir o dataset para um conjunto mais interpretável, sem perder informações relevantes para o problema. Nas próximas etapas, ainda vamos verificar valores nulos, tipos de dados e possíveis ajustes antes de definir as variáveis finais para modelagem.

In [22]:
selected_missing = (
    df_selected.isna()
    .sum()
    .sort_values(ascending=False)
    .reset_index()
)

selected_missing.columns = ["column_name", "missing_count"]

selected_missing["missing_percentage"] = (
    selected_missing["missing_count"] / len(df_selected) * 100
).round(2)

selected_missing

,column_name,missing_count,missing_percentage
0,koi_num_transits,1142,11.94
1,koi_teq,363,3.80
2,koi_slogg,363,3.80
3,koi_smass,363,3.80
4,koi_depth,363,3.80
5,koi_model_snr,363,3.80
6,koi_prad,363,3.80
7,koi_srad,363,3.80
8,koi_steff,363,3.80
9,koi_insol,321,3.36


### Análise dos valores ausentes nas variáveis selecionadas

A verificação dos valores ausentes mostra que as variáveis selecionadas possuem um nível de preenchimento adequado para a continuidade do projeto.

A coluna com maior quantidade de valores ausentes é `koi_num_transits`, com 11,94% dos registros vazios. Apesar disso, a variável ainda pode ser útil, pois representa a quantidade de trânsitos observados e pode contribuir para avaliar a consistência do sinal detectado.

Outras variáveis importantes, como `koi_depth`, `koi_model_snr`, `koi_prad`, `koi_steff`, `koi_slogg`, `koi_srad` e `koi_smass`, possuem cerca de 3,80% de valores ausentes. Esse percentual é baixo o suficiente para ser tratado na etapa de limpeza, sem a necessidade de remover essas colunas neste momento.

Com isso, a seleção inicial de variáveis continua válida. Nas próximas etapas, será necessário decidir a melhor estratégia de tratamento para esses valores, como remoção de linhas incompletas ou preenchimento por alguma medida estatística, processo conhecido como `imputation`.

In [23]:
df_selected.dtypes

kepid                 int64
kepoi_name              str
koi_disposition         str
koi_period          float64
koi_duration        float64
koi_depth           float64
koi_num_transits    float64
koi_model_snr       float64
koi_prad            float64
koi_teq             float64
koi_insol           float64
koi_steff           float64
koi_slogg           float64
koi_srad            float64
koi_smass           float64
koi_kepmag          float64
ra                  float64
dec                 float64
dtype: object

### Análise dos tipos de dados das variáveis selecionadas

Os tipos de dados estão adequados para esta etapa do projeto.

As colunas `kepid`, `kepoi_name` e `koi_disposition` aparecem como variáveis de identificação ou classificação. A coluna `koi_disposition` será usada como variável-alvo, ou seja, a informação que o modelo tentará prever.

As demais colunas principais, como `koi_period`, `koi_duration`, `koi_depth`, `koi_model_snr`, `koi_prad`, `koi_steff`, `koi_slogg`, `koi_srad`, `koi_smass`, `ra` e `dec`, estão em formato numérico. Isso é positivo, porque modelos de Machine Learning e Deep Learning precisam receber variáveis numéricas como entrada.

In [25]:
df_selected.to_csv("../data/processed/koi_selected_initial.csv", index=False)

print("Arquivo salvo em: ../data/processed/koi_selected_initial.csv")

Arquivo salvo em: ../data/processed/koi_selected_initial.csv


### Análise do salvamento da seleção inicial

O arquivo `koi_selected_initial.csv` foi salvo na pasta `data/processed`, contendo apenas as variáveis selecionadas para a primeira versão da análise.

Esse salvamento é importante porque separa o dataset bruto da versão inicial tratada para o projeto. Assim, mantemos uma organização mais clara: o arquivo original permanece preservado, enquanto a versão selecionada pode ser usada nas próximas etapas de exploração e preparação dos dados.

### Conclusão parcial da validação do dataset

A validação inicial mostra que o dataset escolhido é adequado para o desenvolvimento do projeto **ExoQuantum AI**.

A base possui 9.564 registros e 153 colunas originalmente. Após a inspeção inicial, foi criada uma versão reduzida com variáveis de identificação, classificação e características físicas/observacionais relacionadas ao trânsito, ao possível planeta e à estrela hospedeira.

A variável `koi_disposition` foi identificada como a coluna-alvo do projeto, pois classifica os objetos em `FALSE POSITIVE`, `CONFIRMED` e `CANDIDATE`. Essa estrutura permite formular o problema como uma tarefa de classificação supervisionada.

A análise também mostrou que existe um leve desbalanceamento entre as classes, conhecido como `class imbalance`. Esse ponto será considerado nas próximas etapas, principalmente na escolha das métricas de avaliação e no treinamento dos modelos.

Em relação à qualidade dos dados, as variáveis selecionadas apresentam um nível aceitável de valores ausentes. Nenhuma das colunas escolhidas precisa ser descartada neste momento apenas por falta de preenchimento. Os valores faltantes serão tratados em uma próxima etapa, usando técnicas adequadas de limpeza e `imputation`.

Também foi identificado um possível risco de `data leakage` em colunas diretamente relacionadas à marcação de falsos positivos. Por isso, essas variáveis não foram incluídas na seleção inicial, mantendo o projeto mais consistente e evitando que os modelos aprendam por atalhos.

Com base nessa validação, o dataset será mantido como base principal do projeto. A próxima etapa será realizar uma análise exploratória mais detalhada, buscando entender melhor os padrões das classes e das variáveis selecionadas.